zerver/tests/test_retention.py

```diff
@@ -885,6 +886,45 @@ def test_get_realms_and_streams_for_archiving(self) -> None:
         self.assert_length(result, len(simple_algorithm_result))
         self.assertEqual(result, simple_algorithm_result)
 
+class TestRestoreStreamMessages(ArchiveMessagesTestingBase):
+    def test_restore_retention_policy_deletions_for_stream(self) -> None:
+        cordelia = self.example_user('cordelia')
+        hamlet = self.example_user('hamlet')
+
+        realm = get_realm("zulip")
+        stream_name = "Denmark"
+        stream = get_stream(stream_name, realm)
+
+        message_ids_to_archive_manually = [
+            self.send_stream_message(cordelia, stream_name, str(i)) for i in range(0, 10)
+        ]
+        usermessage_ids_to_archive_manually = self._get_usermessage_ids(message_ids_to_archive_manually)
+        message_ids_to_archive_by_policy = [
+            self.send_stream_message(hamlet, stream_name, str(i)) for i in range(0, 10)
```

Comment:
Do we need to send 10 messages each?  Might be better to do e.g. 2 each; otherwise, we're potentially wasting a bit of time in the test suite.

Reply:
Yup, let's do 2